Projects/Contracts/Sources/View/ContractTable.swift

```diff
@@ -142,3 +154,37 @@ struct ContractTable: View {
         }
     }
 }
+
+@MainActor
+public class ContractTableViewModel: ObservableObject {
+    @Published var viewState: ProcessingState = .loading
+    @PresentableStore var store: ContractStore
+    @Published var actionCancellable: AnyCancellable?
+    @Published var loadingCancellable: AnyCancellable?
+
+    init() {
+        actionCancellable = store.actionSignal
```

Comment:
I think you dont need actionCancellable and following that actionSignal.
Only having this should work
```
        loadingCancellable = store.loadingSignal
            .receive(on: RunLoop.main)
            .sink { _ in
            } receiveValue: { [weak self] action in
                let getAction = action.first(where: { $0.key == .fetchContracts })
                switch getAction?.value {
                case let .error(errorMessage):
                    self?.viewState = .error(errorMessage: errorMessage)
                case .loading:
                    self?.viewState = .loading
                default:
                    self?.viewState = .success
                }
            }
```